flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/ResourceActions.java

```diff
@@ -42,9 +40,9 @@
 	 * Requests to allocate a resource with the given {@link ResourceProfile}.
 	 *
 	 * @param resourceProfile for the to be allocated resource
-	 * @return Collection of {@link ResourceProfile} describing the allocated slots
+	 * @return whether the resource can be allocated
 	 */
-	Collection<ResourceProfile> allocateResource(ResourceProfile resourceProfile);
+	boolean allocateResource(ResourceProfile resourceProfile);
```

Comment:
Wouldn't it be required for future developments that we can tell `allocateResource` how big resource-wise the worker should be?

Reply:
I'm not sure whether I have correctly understood your question.
Telling `allocateResource` how big the worker is in the scope of making slot allocation pluggable. In this PR, we have the SM to tell `allocateResource` the `WorkerResourceSpec`, in a subsequence commit rather than this one. This PR does not make the RM implementations to respect the `WorkerResourceSpec` they are told. We make those changes in subsequence PRs.

Reply:
All right, understood.